spec/takeaway_spec.rb

```diff
@@ -0,0 +1,66 @@
+require 'takeaway_1'
+
+describe TakeAway do
+  subject(:takeaway) { described_class.new }
+
+  let(:dishes) do
+    {
+      hotdog: 5.00,
+      fries: 3.50
+    }
+  end
+
+  let(:order) do
+    {
+      hotdog: 2,
+      fries: 1
+    }
+  end
+
+  it "creates a new takeaway" do
+    expect(takeaway).to be_a_kind_of TakeAway
+  end
+
+  it "has a list of dishes with prices" do
+    expect(takeaway.dishes).to eq(dishes)
+  end
+
+  it "shows the menu when called" do
+    printed_menu = "Hotdog £5.00, Fries £3.50"
+    expect(takeaway.print_menu).to eq(printed_menu)
+  end
+
+  it "selects several dishes from the menu" do
+    takeaway.add(:hotdog, 2)
+    takeaway.add(:fries, 1)
+    expect(takeaway.order).to eq(order)
+  end
+
+  it "calculates the total for the order" do
+    takeaway.add(:hotdog, 2)
+    takeaway.add(:fries, 1)
+    total = 13.50
+    expect(takeaway.total).to eq(total)
+  end
+
+  it "tells if a dish is on the menu" do
+    expect(takeaway.has_dish?(:hotdog)).to be true
+  end
+
+  it "tells if a dish is not on the menu" do
+    expect(takeaway.has_dish?(:hamburger)).to be false
+  end
+
+  it "doesn't allow items to be added that are not on the menu" do
```

Comment:
I would imagine this spec would pass however its not the norm to stub part of a class under test. See an explanation at https://softwareengineering.stackexchange.com/questions/107361/is-it-okay-to-fake-part-of-the-class-under-test

Rather consider making your menu a dependency of TakeAway. You could then test variants of your menu without stubbing.

e.g 

```
menu_hotdog_and_fries = { 
  hotdog: 5.00,
  fries: 2.00
}

menu_hotdog = {
  hotdog: 5
}

takeaway_one = TakeAway.new(menu_hotdog_and_fries)
takeaway_two = TakeAway.new(menu_hotdog)

your 'menu' could be a simple hash or if you really need a class in its own right.